matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/send/LocalEchoEventFactory.kt

```diff
@@ -507,13 +507,13 @@ internal class LocalEchoEventFactory @Inject constructor(
         return createMessageEvent(roomId, content)
     }
 
-    private fun createMessageEvent(roomId: String, content: MessageContent? = null): Event {
-        return createEvent(roomId, EventType.MESSAGE, content.toContent())
+    private fun createMessageEvent(roomId: String, content: MessageContent? = null, currentEventId: String? = null): Event {
+        return createEvent(roomId, EventType.MESSAGE, content.toContent(), currentEventId)
     }
 
-    fun createEvent(roomId: String, type: String, content: Content?): Event {
+    fun createEvent(roomId: String, type: String, content: Content?, currentEventId: String? = null): Event {
```

Comment:
It feels a bit hacky to use an eventId on the LocalEchoEventFactory, it should only gives Event with local.id.
I don't think we need a complete new Event but just a new Content. Create a method which returns the Content and uses this in both the Timeline and this Factory 
```
MessageTextContent(
                msgType = MessageType.MSGTYPE_TEXT,
                format = MessageFormat.FORMAT_MATRIX_HTML,
                body = replyFallback,
```